[ 할 일 ] 할 일 생성/수정 목표 선택 DropdownMenu로 변경

MinaRoh opened the pull request:
## ✅ 작업 내용
할 일 모달에서 목표선택 input select를 공통컴포넌트 DropdownMenu로 변경했습니다.

- DropdownMenu 커스텀 가능하게 수정
- 할 일 추가 모달에서 목표 선택 DropdownMenu로 변경하기
- 할 일 수정 모달에서 목표 선택 DropdownMenu로 변경하기
- 새 할일 제목 입력 후 목표 클릭시 폼 전송되는 오류 수정
- 새 할일 폼 제출 후 남아있던 이전 기록 초기화
- DropdownMenu 최대 높이 설정해 모달 밖으로 빠져나가지 않도록 수정


## 📸 스크린샷 / GIF / Link
before
![image](https://github.com/user-attachments/assets/50d87b80-eeb2-40ed-844f-e87761940870)

after
![image](https://github.com/user-attachments/assets/2e0df497-8e37-4291-b21a-e3d2be3eefca)

![image](https://github.com/user-attachments/assets/ae7bc948-7cec-495c-8317-0b1091585d69)

MinaRoh committed:
✨ #92 feat: 새 할일 모달 목표선택 dropdown 구현

- input select type으로 되어 있던 목표 선택을 DropdownMenu 공통컴포넌트로 변경했습니다.
- DropdownMenu를 보편적으로 사용 할 수 있게 수정했습니다.

MinaRoh committed:
🐛 #92 fix: 새 할일 제목 입력 후 목표 클릭시 폼 전송되는 오류 수정

MinaRoh committed:
🐛 #92 fix: 새 할일 폼 제출 후 남아있던 이전 기록 초기화

- 할 일 추가 후 한 번 더 새 할일을 클릭 했을 때 이전에 썼던 할 일의 데이터가 남아있는 오류를 수정했습니다.

MinaRoh committed:
✨ #92 feat: 할 일 수정 모달 목표 선택 dropdownMenu로 변경

MinaRoh committed:
🩹 #92 fix: 주석 제거

ahrrrl reviewed (approved):
![02_모코코콘2_28_신나](https://github.com/user-attachments/assets/f47e14ec-f210-435a-af8f-116134ea10a4)

MinaRoh commented:
아 드롭다운에서 항목들이 가운데 정렬이 아니고 왼쪽 정렬이네요 수정하겠습니다

ahrrrl commented on components/common/DropdownMenu.tsx:
```diff
@@ -46,28 +64,37 @@ const DropdownMenu = ({ icon: Icon, dropdownList, onItemClick, className: iconBu
-        >
-          <ul className='flex flex-col text-nowrap text-sm sm:text-lg lg:text-lg text-slate-700 justify-center items-center text-center'>
+        <div ref={dropdownRef} className={dropdownClassNames}>
+          <ul className='flex flex-col text-nowrap text-sm sm:text-lg lg:text-lg justify-center items-center text-center'>
```
혹시 드롭다운 리스트 최대 길이 설정해주실 수 있나요? 목표가 20개 있으면 막커져서 별로일거같아요!

MinaRoh replied:
오 네 좋은것같네요 max 정해두고 스크롤하게 하겠습니다

MinaRoh committed:
💄 #92 style: 디자인 반영

- 드롭다운 메뉴 텍스트를 가운데 정렬에서 왼쪽 정렬로 수정했습니다.
- 상단에 '목표' 텍스트 라벨을 추가했습니다.
- 드롭다운 메뉴의 최대 길이를 설정해 모달을 벗어나지 않도록 수정했습니다.
